src/main/scala/uncore/tilelink2/RationalCrossing.scala

```diff
@@ -0,0 +1,177 @@
+// See LICENSE.SiFive for license details.
+
+package uncore.tilelink2
+
+import Chisel._
+import chisel3.internal.sourceinfo.SourceInfo
+import config._
+import diplomacy._
+import util._
+
```

Comment:
To me, "Rational" is not an intuitive name for this, but that's not my complaint. Can you please add a comment explaining what "Rational" means (basically what you said in this PR)